src/main.cpp

```diff
@@ -4451,7 +4451,7 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
             assert(recentRejects);
             recentRejects->insert(tx.GetHash());
 
-            if (pfrom->fWhitelisted) {
+            if (pfrom->fWhitelisted && GetBoolArg("-whitelistalwaysrelay", true)) {
```

Comment:
Please define a `static const bool DEFAULT_WHITELISTALWAYSRELAY = true;` in `main.h`  and use that
(sorry for making your patch even more hideously huge)